meta-boxes.php

```diff
@@ -1,20 +1,139 @@
 <?php
+/**
+ * File containing meta box callback functions
+ */
 
-function nprstory_publish_meta_box($post) {
-	$helper_text = 'Push this story to NPR:';
-	$is_disabled = ($post->post_status != 'publish');
-	if ($is_disabled) {
-		$helper_text = 'Publish this story in order to push it to NPR.';
+/**
+ * Output the NPR Story API publishing options metabox for the edit page admin interface
+ *
+ * @param WP_Post $post the WordPress post object.
+ * @see nprstory_save_send_to_api
+ * @see nprstory_save_send_to_org
+ * @see nprstory_save_send_to_one
+ * @see nprstory_save_nprone_featured
+ *
+ * @todo When there is better browser support for input type="datetime-local", replace the jQuery UI and weird forms with the html5 solution. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local
+ */
+function nprstory_publish_meta_box( $post ) {
+	$helper_text = __( 'Push this story to NPR:', 'nprapi' );
+	$is_disabled = ( 'publish' !== $post->post_status );
+	$attrs = array( 'id' => 'ds-npr-update-push' );
+
+	if ( $is_disabled ) {
+		$helper_text = __( 'Publish this story in order to push it to NPR.', 'nprapi' );
 	}
-	$attrs = array('id' => 'ds-npr-update-push');
-	if ($is_disabled)
+
+	if (  $is_disabled ) {
 		$attrs['disabled'] = 'disabled';
-?>
+	}
+
+	wp_enqueue_style( 'jquery-ui' );
+	wp_enqueue_style( 'nprstory_publish_meta_box_stylesheet' );
+	wp_enqueue_script( 'nprstory_publish_meta_box_script' );
+
+	?>
 	<div id="ds-npr-publish-actions">
-		<p class="helper-text"><?php echo $helper_text; ?></p>
-<?php
-	submit_button( 'Push to NPR', 'large', 'ds_npr_update_push', false, $attrs );
-?>
+		<ul>
+		<?php
+			// send to the npr api
+			// The meta name here is '_send_to_nprone' for backwards compatibility with plugin versions 1.6 and prior
+			$nprapi = get_post_meta( $post->ID, '_send_to_nprone', true ); // 0 or 1
+			if ( '0' !== $nprapi && '1' !== $nprapi ) { $nprapi = 1; } // defaults to checked; unset on new posts
+
+			// this list item contains all other list items, because their enabled/disabled depends on this checkbox
+			echo '<li>';
+			printf(
+				'<label><input value="1" type="checkbox" name="send_to_api" id="send_to_api" %2$s/> %1$s</label>',
+				__( 'Send to NPR API', 'nprapi' ),
+				checked( $nprapi, '1', false )
```

Comment:
I don't know what the function checked() does - why are there two parameters a '1'  and a false? - so just confirm that this is setting it to checked by default per the SOW.

Reply:
[`checked()`](https://developer.wordpress.org/reference/functions/checked/) is a WordPress built-in function that in this case outputs the HTML `checked="checked"` if the value in `$nprapi` is equal to the default value, which is the string `'1'`. The `false` parameter here just means that the function's output should be returned instead of `echo`ed; this is done because the output of `checked()` is being used as a parameter on `printf()` for inclusion in the string `'<label><input value="1" type="checkbox" name="send_to_api" id="send_to_api" %2$s/> %1$s</label>'` at the point  `%2$s`. 

The default value of `$nprapi` is set a few lines up from here, at this line:

```php
$nprapi = get_post_meta( $post->ID, '_send_to_nprone', true ); // 0 or 1
if ( '0' !== $nprapi && '1' !== $nprapi ) { $nprapi = '1'; } // defaults to checked; unset on new posts
```

[`get_post_meta()`](https://developer.wordpress.org/reference/functions/get_post_meta/) doesn't provide a way to have a default value. When called in the post editor for a post that doesn't exist yet, or for a post that doesn't have this meta key saved, `get_post_meta()` will return `false`. Because the boolean `false` is not type-strictly equal to the strings `'0'` or `'1'`, the conditional here sets `$nprapi` to `'1'`.

If `get_post_meta()` returns a value that is `'0'` or `'1'`, then we use that value.

`checked()` says the checkbox is checked if `$nprapi` is `'1'`, so the checkbox defaults to checked because `$nprapi` defaults to a value that means the checkbox should be checked.

In the event of something weird being saved as this post meta, `$nprapi` would still be set to `'1'`. Saving the post calls the function `nprstory_save_send_to_api()` which would overwrite the weird value, or an unset value, with the value of the checkbox. If the user hasn't unchecked the box, then the saved value will be that the box should be checked.

(Sorry for really long explanation; I rubber-ducked here and found a bug as a result.)

Reply:
Thank you for the explanation!